Pages/CodeButton.js

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useContext } from "react";
+import {
+  Text,
+  View,
+  StyleSheet,
+  TextInput,
+  Button,
+  Alert,
+  Modal,
+  TouchableHighlight,
+} from "react-native";
+import { useMutation, gql } from "@apollo/client";
+import { useForm, getErrors } from "../util/hooks";
+import Constants from "expo-constants";
+
+function CodeButton(props) {
+  const [errors, setErrors] = useState({});
+  const [modalVisible, setModalVisible] = useState(false);
+
+  let user = props.user;
+
+  const { values } = useForm(redeemPointsCallback, {
+    code: "",
+    username: user.username,
+  });
+
+  function redeemPointsCallback() {
+    redeemPoints();
+  }
+
+  const [redeemPoints] = useMutation(REDEEM_POINTS_MUTATION, {
+    update(_, { data: { redeemPoints: userData } }) {
+      values.code = "";
+      setErrors(false);
+      setModalVisible(false);
+      updateGetUser(userData);
+    },
+
+    onError(err) {
+      getErrors(err);
+    },
+
+    variables: values,
+  });
+
+  function updateGetUser(userData) {
+    user.fallPoints = userData.fallPoints;
+    user.springPoints = userData.springPoints;
+    user.summerPoints = userData.summerPoints;
+    user.events = userData.events;
+    user.tasks = userData.tasks;
+    user.message = userData.message;
+  }
+
+  return (
+    <View style={styles.container}>
+      <Modal
+        animationType="slide"
+        transparent={true}
+        visible={modalVisible}
+        onRequestClose={() => {
+          setModalVisible(false);
+          Alert.alert("Modal has been closed.");
+        }}
+      >
+        <View style={styles.modalView}>
+          <Text style={styles.header}>Redeem Points</Text>
+          <Text style={styles.label}>Enter code:</Text>
+          <TextInput
+            style={styles.input}
+            placeholder="Code..."
+            onChangeText={(value) => (values.code = value)}
+            spellCheck={false}
+            autoCorrect={false}
+          />
+          <TouchableHighlight
+            style={styles.button}
+            onPress={() => redeemPointsCallback()}
+          >
+            <Text style={styles.textStyle}>Submit</Text>
+          </TouchableHighlight>
+
+          <TouchableHighlight
+            style={styles.button}
+            onPress={() => {
+              setModalVisible(!modalVisible);
+            }}
+          >
+            <Text style={styles.textStyle}>Cancel</Text>
+          </TouchableHighlight>
+        </View>
+      </Modal>
+
+      <TouchableHighlight
```

Comment:
why the touchable highlight instead of the button? https://reactnative.dev/docs/button